src/Features/Core/Portable/CodeRefactorings/UseNamedArguments/AbstractUseNamedArgumentsCodeRefactoringProvider.cs

```diff
@@ -34,7 +34,17 @@ protected abstract class Analyzer<TBaseArgumentSyntax, TArgumentSyntax, TArgumen
                     return;
                 }
 
-                var argument = root.FindNode(context.Span).FirstAncestorOrSelf<TBaseArgumentSyntax>() as TArgumentSyntax;
+                var token = root.FindToken(context.Span.Start);
+                if (IsCloseParenOrComma(token))
+                {
+                    if (token.Span.Start != context.Span.Start ||
+                        IsCloseParenOrComma(token = token.GetPreviousToken()))
```

Comment:
You mean the inline assignment or duplicated check for comma? I could inline the whole FirstAncestorOrSelf and bail when we reach an argument list instead. thoughts?